src/main/resources/mutations.graphqls

```diff
@@ -0,0 +1,5 @@
+type Mutation {
+    startRecordingByNodes(recording: RecordingSettings!, nodes: [NodeSelector!]!): [TargetNode!]!
+    startRecordingByAnnotations(recording: RecordingSettings!, annotations: [AnnotationSelector!]!): [TargetNode!]!
+    startRecordingByLabels(recording: RecordingSettings!, labels: [LabelSelector!]!): [TargetNode!]!
```

Comment:
>  TODO - can these all be done in one query? With an implied "AND" or "OR"
    of the various selector parameters? Or would the AND/OR be another argument?
    targetsWithLabel(key: String!, value: String): [TargetNode!]!
    targetsWithAnnotation(key: String!, value: String): [TargetNode!]!
    targetsWithAnnotations(annotations: [String!]!): [TargetNode!]!

Similar to the Queries, if a user can query for targets with annotations `AND`/`OR` labels, would it make sense for users to be able to start/stop recordings with the same amount of flexibility to select by annotation and/or label? How would the mutation definitions need to change?

Reply:
IMO what makes the most sense is that the mutation definitions should generally mirror the query definitions, except they also require this `RecordingSettings` parameter. Given a Selector, or some series of Selectors and some Operation, we can expect the Query to tell us about all of the targets or recordings matching the Selector(s)/Operation, or we can expect the Mutation to start a recording using the given settings on all targets matching the Selector(s)/Operation.

The nice thing GraphQL gives us is that the Query side can always be defined as telling the client about the Targets, but that the Targets also contain reference to the Recordings, and GraphQL knows how to resolve the Recordings. If the user only cares about the Targets then they just make their query for that level of detail and that's what they get back, and the server does no unnecessary work. If they do need to know about the recordings then they add that request into their query and the GraphQL engine figures out how to call the resolver and prepare the result, only causing the server to do the extra work of connecting to the targets (or consulting the in-memory cache as described earlier) to populate the active recordings list for each target if that's actually requested and required.